aws/resource_aws_appsync_datasource_test.go

```diff
@@ -384,6 +385,67 @@ func TestAccAwsAppsyncDatasource_Type_None(t *testing.T) {
 	})
 }
 
+func TestAccAwsAppsyncDatasource_Type_RelationalDatabase(t *testing.T) {
+	oldRegion := os.Getenv("AWS_DEFAULT_REGION")
+	os.Setenv("AWS_DEFAULT_REGION", "us-east-1")
+	defer os.Setenv("AWS_DEFAULT_REGION", oldRegion)
```

Comment:
Why are you overriding the region here? Perhaps only running the test if the user-provided region has the service available is a better idea, and more consistent with other tests. i.e. changing lines 396 and 428 to `		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(appsync.EndpointsID, t) },`